### DISCUSSION TITLE
JSON issue with Python client, or am I doing something wrong?

### DISCUSSION THREAD

**Discussion by makarr** [duckdb/duckdb#12558] [General]
Here is a minimal example using the Python client, duckdb version 1.0.0:

```
import duckdb

metadata = {"bool_val": True, "int_val": 42, "str_val": "x"}
con = duckdb.connect()
con.sql("CREATE TABLE data (id UBIGINT, metadata JSON)")
con.execute("INSERT INTO data VALUES (?, ?)", [1, metadata])

con.execute("SELECT id FROM data WHERE metadata.bool_val = ?", [True]).fetchone() # (1,)
con.execute("SELECT id FROM data WHERE metadata.int_val = ?", [42]).fetchone() # (1,)

con.execute("SELECT id FROM data WHERE metadata.str_val = ?", ["x"]).fetchone() # error
con.execute("SELECT id FROM data WHERE metadata.str_val = 'x'").fetchone() # error
```

The error message is:
Conversion Error: Malformed JSON at byte 0 of input: unexpected character.  Input: x

Is this a bug or is there something I'm not formatting correctly?

**Comment by makarr** [comment]
OK I figured this out, thankfully. To get it to work it's necessary to wrap the string in another set of quotation marks like so:

`con.execute("SELECT id FROM data WHERE metadata.str_val = ?", ['"x"']).fetchone()`

**Reply by Tishj** [reply]
That is not related to Python, you should use the right method to extract the string value from the json

<https://duckdb.org/docs/extensions/json#json-extraction-functions>

**Reply by makarr** [reply]
Great, thank you, I missed the the `json_extract_string` method
